### DISCUSSION TITLE
Starter Prompts with If/else function

### DISCUSSION THREAD

**Discussion by amansoniamazatic** [FlowiseAI/Flowise#1643] [General]
Flowise currently offers the option to include starter prompts. However, I am facing an issue where I am unable to use starter prompts directly within an if/else condition. To work around this, I have to first employ an LLM chain, which I find inconvenient. I suggest adding functionality that allows for the direct use of if/else conditions from the chat input as an input variable. This way, decisions can be made based on the text, determining whether it should be passed to the LLM or not, all within the if/else condition. I believe this approach will save both time and money. Please consider implementing this update in Flowise.

**Comment by Jaredude** [comment]
Maybe I'm misunderstanding your request, but here's a flow that has no LLM at all. If/Else has access to the question that comes from the starter prompt and branches accordingly.

https://github.com/FlowiseAI/Flowise/assets/10228950/b350c3a2-765c-4952-8ba2-2f511459c310

**Reply by amansoniamazatic** [reply]
This is the thing i want thanks also @Jaredude i am curious if "if" condition satisfy then it goes to custom js function which i want but if the condition does't satisfy and it returns false at that case i want the question should goes to conversational retrieval QA chain and continue the conversation can you please provide a demo of this also.

**Reply by Jaredude** [reply]
The output of the if/else node has to be string | number | boolean | json | array, which makes it connectable to quite a few nodes, but a Conversation Retrieval Agent is currently not a node you can connect to. You can easily connect an LLM chain flow, but you cannot directly connect to a CRA. At least not currently, and not directly.

A work around (not really) would be to write a call to a separate Q&A flow within your custom JS function and pass the question on that call. The reason this isn't a good work around is that with every call to the Q&A flow, it would be creating a new sessionId for every question that is asked on that flow. So, while it can be done, it shouldn't be done.

**Reply by kzmKZuzmo** [reply]
hey @Jaredude how did you create this buttons ? 

![image](https://github.com/FlowiseAI/Flowise/assets/16986250/698b7552-a4fd-4c95-a5e7-7e26aa7c46ce)

**Reply by amansoniamazatic** [reply]
@kzmKZuzmo This is starter prompts you can create them by going to configuration then starter prompts.
![Screenshot from 2024-05-02 15-12-59](https://github.com/FlowiseAI/Flowise/assets/153904988/14c8a976-3db7-44db-b756-0d3392c22766)

**Reply by nerosguest** [reply]
> The output of the if/else node has to be string | number | boolean | json | array, which makes it connectable to quite a few nodes, but a Conversation Retrieval Agent is currently not a node you can connect to. You can easily connect an LLM chain flow, but you cannot directly connect to a CRA. At least not currently, and not directly.
> 
> A work around (not really) would be to write a call to a separate Q&A flow within your custom JS function and pass the question on that call. The reason this isn't a good work around is that with every call to the Q&A flow, it would be creating a new sessionId for every question that is asked on that flow. So, while it can be done, it shouldn't be done.

@Jaredude can you explain to me how to execute this method to connect CRA to if/else node. can't seem to understand this. very new to all this so it would help if you dumb down this.

**Reply by Jaredude** [reply]
@nerosguest Sorry if it wasn't clear in my response, but you **cannot** connect a CRA and if/else node.

**Comment by Jaredude** [comment]
@kzmKZuzmo The answer from @amansoniamazatic will get you what you need

> @kzmKZuzmo This is starter prompts you can create them by going to configuration then starter prompts. ![Screenshot from 2024-05-02 15-12-59](https://private-user-images.githubusercontent.com/153904988/327384954-14c8a976-3db7-44db-b756-0d3392c22766.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2NjIzMjksIm5iZiI6MTcxNDY2MjAyOSwicGF0aCI6Ii8xNTM5MDQ5ODgvMzI3Mzg0OTU0LTE0YzhhOTc2LTNkYjctNDRkYi1iNzU2LTBkMzM5MmMyMjc2Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwMlQxNTAwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ODZiZTU5NjU0OTJhZmI3NGIyYjIzZjhiYmM3ODI4ODViNjhmMDJjNjM1YTNmYmVlMDE1MzMwNzM3MjQ5ZDIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3NfYNEx8zG9Lj-vrZp_6Y-WWJVSjsdeyp2vO6-pHIwM)
